pagopa/io-app
# feat: [IOCOM-872] Base F24 section with new DS for SEND messages

## Short description
This PR adds the F24 section to a SEND message, using the new DS.
The F24 bottom sheet will come later.
|One F24|Multiple F24|
|-|-|
|![Simulator Screenshot - iPhone 15 - 2024-03-15 at 15 42 20](https://github.com/pagopa/io-app/assets/5150343/95ba1cf4-0760-43dd-a21d-c871f9efd3ca)|![Simulator Screenshot - iPhone 15 - 2024-03-15 at 15 42 30](https://github.com/pagopa/io-app/assets/5150343/07dea816-35e2-42ec-b8af-e269c2d79950)|

## List of changes proposed in this pull request
- New `F24` component that displays the F24 sections, with business logic to:
  -  hide it if the message is cancelled or there are no F24
  - show an attachment cell if there is only an F24 and handle its download and displaying
  - show a link if there is more than a single F24 (its CTA is not implemented yet)
- `MessageDetails` uses `F24` to display the F24 section
- `MessageF24` renamed to `LegacyMessageF24`

## How to test
Using the io-dev-api-server, check that a SEND message:
- does not show the F24 section if the message is cancelled, even if it has F24s
- does not show the F24 section if the message has no F24s
- shows the F24 section with one attachment cell if the message has just one F24
- shows the F24 section with the link and no attachment cells if the message has more than one F24
